game-core/src/main/java/games/strategy/net/AbstractServerMessenger.java

```diff
@@ -260,32 +212,60 @@ private void scheduleMacUnmuteAt(final String mac, final Instant checkTime) {
   public void notifyPlayerLogin(final String uniquePlayerName, final String mac) {
     synchronized (cachedListLock) {
       cachedMacAddresses.put(uniquePlayerName, mac);
-      if (isLobby()) {
```

Comment:
The code below was previously only executed for a lobby server; now it is executed for both a lobby server and a game server.  However, the game server path will have no side effects because the `getUsernameUnmuteTime()` and `getMacUnmuteTime()` methods return an empty `Optional` for `GameServerMessenger`.

Thus, other than a few additional `if` checks being executed that weren't executed prior to this PR, it is effectively the same code.

Reply:
I'm wondering: Does a GameServer even implement a muting functionality?
If not the code could simply be extracted to the lobby code, so the other implementations never even get the chance to execute it.

Reply:
Yes, the game host can mute a player from the **Network** menu in a headed game server.  In a headless game server, a lobby moderator can mute a player from the bot context menu in the lobby client.

Reply:
@ssoloff  Sorry to interrupt but does this mean we can now mute players in a bot?  We used to have to boot them I thought?

Reply:
@prastle AFAIK, this feature is in the current release.  If you right-click a bot in the lobby (as a moderator), do you not see a **Mute Player in Headless Host Bot** menu item?  When I tested this locally it appears to work as one would expect.  (Although, specifying `0` to unmute a user, as the instructions indicate, will result in an exception due to how the code calculates time offsets.)

Reply:
You are correct I never realized it finally worked 👍

Reply:
@prastle Yes, lobby mute is totally independent of in-game mute.